### ISSUE TITLE
Automatically pick the right config based on the type of map being played

### ISSUE THREAD

**Issue by icewind1991** [Arie/serveme#20]
It's common for people when switching map types (pl, koth, 5cp) to forget to execute the right config for the new map leading to various issues while playing.

It would be nice if the server would automatically switch the config when loading a map.

This could be done with a server plugin like https://github.com/spiretf/autoexec

**Comment by cob16** [comment]
Do you know if that plugin handles having mutable league configs?

**Comment by icewind1991** [comment]
Updated the plugin to automatically set the used league and gamemode when manually changing config

So when loading `etf2l_9v9_koth` and then switching to `cp_gullywash` to will stay on etf2l hl config (`etf2l_9v9_5cp`)

**Comment by cob16** [comment]
Ah thats nice, and friendly user messages too! 

@Arie your thoughts?

**Comment by Arie** [comment]
Yeah cool, I'm a bit hesitant about the auto-loading of a config. There's no single right config for a given map. You might play UGC or ETF2L for example.

I like the automated switching of config based on the current config+map though.

**Comment by cob16** [comment]
@Arie:  Just to note as per icewind's last comment the plugin handles league types automatically 

I think this is harmless to have as Servme sets the starting config on the reservations page allready. This plugin just adds convenience for the users when they chance maps in scrims saving them the cvar (something that half the tf2 population don't remember how to do 😄 ). I doubt that users often want to change between leagues in there reservations so it's a better default than a PL config for a KOTH map. 

I you feel it's necessary I could always drum up a pull request to make it an option on the reservation page

**Comment by AndyBarnett** [comment]
:+1: for this plz!
